lib/ecto/adapters/mysql/connection.ex

```diff
@@ -504,22 +504,29 @@ if Code.ensure_loaded?(Mariaex) do
 
     defp create_names(prefix, sources, pos, limit) when pos < limit do
       current =
-        case elem(sources, pos) do
-          {table, schema} ->
-            name = [create_alias(table) | Integer.to_string(pos)]
-            {quote_table(prefix, table), name, schema}
-          {:fragment, _, _} ->
-            {nil, [?f | Integer.to_string(pos)], nil}
-          %Ecto.SubQuery{} ->
-            {nil, [?s | Integer.to_string(pos)], nil}
-        end
+        sources
+        |> elem(pos)
+        |> create_name(prefix, pos)
```

Comment:
It seems that, when preparing the queries, we already do `|| query.prefix`. This means we don't need to pass the `query.prefix` here. Shall we remove it from here then?

Reply:
I'm okay with that.